app/models/product.rb

```diff
@@ -12,6 +13,26 @@ class Product < ApplicationRecord
     self.discount_price = price if discount_price.blank?
   end
 
+  after_create_commit do
```

Comment:
do not pass block in the callbacks. create a method for it. DO NOT USE this practice to pass a block in callbacks